battenberg/cli.py

```diff
@@ -55,30 +56,40 @@ def main(ctx, o: str, verbose: bool):
 
 @main.command()
 @click.argument('template')
+@click.option(
+    '--initial-head',
+    help='The initial branch name to use when creating a new repo',
+    default='main'
```

Comment:
any chance we can be reliant on https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch rather than us imposing the default?

Reply:
We can, but looks like still if it's not set we'd have to be opinionated on the default as it looks like `git config` doesn't store the default value :/

```bash
 ✘  ~/Code/py-ml-gradle-plugin   alexla/AIPO-1004  git config --get init.defaultBranch
 ✘  ~/Code/py-ml-gradle-plugin   alexla/AIPO-1004 
```

I say we stick to hard-coding as we want an opinionated mechanism really to drive adoption of the new terminology and can always update this later on if it is more nuanced. I really cannot see this particular default causing many issues really but maybe I'm wrong.

Reply:
while we want to drive the adoption there may be real world scenarios where this breaks down, such as inheriting a shared pipeline template that hardcodes the branch name to `master`, or a release process only allowlisting artifacts following a specific naming convention.

if its not too much work, i'd like it if we could help from redefining "defaults" here and fallback to the users current configuration.

Reply:
Cool think this can be considered complete now with the change now merged from https://github.com/zillow/battenberg/pull/22 :)